src/hooks/screen/useChatScreenHook.ts

```diff
@@ -78,31 +79,45 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
     if (!otherListener[`chat_${selectedChannel?.id}`]) return;
 
     try {
-      queue.addPriorityJob({
-        priority: QueueJobPriority.HIGH,
-        operationLabel: DatabaseOperationLabel.ChatScreen_GetChat,
-        id: selectedChannel?.name,
-        task: async () => {
-          setChannelAsRead(selectedChannel, true);
-          getAllMessages.refetch();
-        }
-      });
-
-      if (type === 'ANONYMOUS') {
+      if (Platform.OS === 'android') {
+        setChannelAsRead(selectedChannel, true);
+        getAllMessages.refetch();
+      } else {
         queue.addPriorityJob({
-          priority: QueueJobPriority.MEDIUM,
-          operationLabel: DatabaseOperationLabel.ChatScreen_GetSelfAnonUserInfo,
+          priority: QueueJobPriority.HIGH,
+          operationLabel: DatabaseOperationLabel.ChatScreen_GetChat,
           id: selectedChannel?.name,
           task: async () => {
-            const userInfo = await UserSchema.getSelfAnonUserInfo(
-              localDb,
-              anonProfileId,
-              selectedChannel?.id
-            );
-            setSelfAnonUserInfo(userInfo);
+            setChannelAsRead(selectedChannel, true);
+            getAllMessages.refetch();
           }
         });
       }
+
+      if (type === 'ANONYMOUS') {
+        if (Platform.OS === 'android') {
+          const userInfo = await UserSchema.getSelfAnonUserInfo(
+            localDb,
+            anonProfileId,
+            selectedChannel?.id
+          );
+          setSelfAnonUserInfo(userInfo);
+        } else {
+          queue.addPriorityJob({
+            priority: QueueJobPriority.MEDIUM,
+            operationLabel: DatabaseOperationLabel.ChatScreen_GetSelfAnonUserInfo,
+            id: selectedChannel?.name,
+            task: async () => {
+              const userInfo = await UserSchema.getSelfAnonUserInfo(
+                localDb,
+                anonProfileId,
+                selectedChannel?.id
+              );
+              setSelfAnonUserInfo(userInfo);
+            }
+          });
+        }
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are for the purpose of differentiating between Android and other platforms. For Android, tasks are executed directly while for other platforms, tasks are added to a priority job queue. This seems like a performance optimization for Android devices. However, it's important to ensure that this change doesn't introduce any race conditions or inconsistencies in the application state across different platforms. If the operations are not dependent on each other and can be executed independently, then this change should be fine.
```diff
+      if (Platform.OS === 'android') {
+        setChannelAsRead(selectedChannel, true);
+        getAllMessages.refetch();
+      } else {
+         queue.addPriorityJob({
+          priority: QueueJobPriority.HIGH,
+          operationLabel: DatabaseOperationLabel.ChatScreen_GetChat,
+           id: selectedChannel?.name,
+           task: async () => {
+            setChannelAsRead(selectedChannel, true);
+            getAllMessages.refetch();
+           }
+         });
+       }
+
+      if (type === 'ANONYMOUS') {
+        if (Platform.OS === 'android') {
+          const userInfo = await UserSchema.getSelfAnonUserInfo(
+            localDb,
+            anonProfileId,
+            selectedChannel?.id
+          );
+          setSelfAnonUserInfo(userInfo);
+        } else {
+          queue.addPriorityJob({
+            priority: QueueJobPriority.MEDIUM,
+            operationLabel: DatabaseOperationLabel.ChatScreen_GetSelfAnonUserInfo,
+            id: selectedChannel?.name,
+            task: async () => {
+              const userInfo = await UserSchema.getSelfAnonUserInfo(
+                localDb,
+                anonProfileId,
+                selectedChannel?.id
+              );
+              setSelfAnonUserInfo(userInfo);
+            }
+          });
+        }
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->